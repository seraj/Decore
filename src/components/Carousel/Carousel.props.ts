import ThemeProps from "../../theme/Theme.props";

export type SwipeDirection = "left" | "down" | "right" | "up" | string;

export default interface CarouselProps
  extends ThemeProps,
    React.HTMLAttributes<HTMLDivElement> {
  onEdge?(swipeDirection: SwipeDirection): void;
  onInit?(): void;
  onLazyLoad?(slidesToLoad: number[]): void;
  onReInit?(): void;
  onSwipe?(swipeDirection: SwipeDirection): void;
  afterChange?(currentSlide: number): void;
  beforeChange?(currentSlide: number, nextSlide: number): void;
  speed?: number;

  autoplay?: boolean;
  autoplaySpeed?: number;
}
