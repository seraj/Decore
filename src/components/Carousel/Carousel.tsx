import * as React from "react";
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import { ThemeContext } from "styled-components";

import StyledCarousel from "./styled/StyledCarousel";
import CarouselProps from "./Carousel.props";

const Carousel: React.FC<CarouselProps> = ({ children, ...props }) => {
  const theme = React.useContext(ThemeContext);

  const decoreSettings = {
    arrows: true,
    dots: true,
    draggable: true,
    swipe: true,
    speed: theme.animations.slider.speed,
    autoplaySpeed: theme.animations.slider.autoplaySpeed
  };

  return (
    <Slider {...decoreSettings} {...props}>
      {children}
    </Slider>
  );
};

export default Carousel;
