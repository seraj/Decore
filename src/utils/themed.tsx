import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
