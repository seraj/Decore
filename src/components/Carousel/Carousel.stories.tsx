import { boolean, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Carousel from "./Carousel";

const stories = storiesOf("Components/Carousel", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => {
    const carousel = (
      <Carousel autoplay={boolean("autoplay", true)}>
        <div>
          <img src="http://placekitten.com/g/400/200" />
        </div>
        <div>
          <img src="http://placekitten.com/g/401/200" />
        </div>
      </Carousel>
    );

    return carousel;
  })
);
