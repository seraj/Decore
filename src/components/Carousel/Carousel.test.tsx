import * as React from "react";

import themed from "../../utils/themed";
import Carousel from "./Carousel";

describe("Carousel", () => {
  it("should render correctly", () => {
    const component = themed(
      <Carousel>
        <div>Test</div>
      </Carousel>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
