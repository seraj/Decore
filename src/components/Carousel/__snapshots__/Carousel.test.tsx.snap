// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render correctly 1`] = `
<div
  className="slick-slider slick-initialized"
>
  
  <div
    className="slick-list"
  >
    <div
      className="slick-track"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "WebkitTransform": "translate3d(0px, 0px, 0px)",
          "WebkitTransition": "",
          "msTransform": "translateX(0px)",
          "opacity": 1,
          "transform": "translate3d(0px, 0px, 0px)",
          "transition": "",
        }
      }
    >
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-current"
        data-index={0}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Test
          </div>
        </div>
      </div>
    </div>
  </div>
  
  
</div>
`;
